### PR TITLE
Feature/switch to gradle

### PR DESCRIPTION
**Pull request by tomaszmichalak**
Migrate to Gradle 5 & Kotlin & Vert.x 3.6.2.

## Description
Migrate:
- from Vert.x 3.5.4 to 3.6.2
- from Maven to Gradle
- from Groovy scripts to Kotlin scripts

## Motivation and Context
We migrate from Maven to Gradle. In Gradle 5 we can switch from Groovy to Kotlin so IDE support is much better. Additionally we can use the native BOM support and better annotation processors support.
It requires also Vert.x version update.

## Upgrade notes (if appropriate)
Not required.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] My code follows the [code style](/CONTRIBUTING.md#coding-conventions) of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

---
I hereby agree to the terms of the Knot.x Contributor License Agreement.

### CODE REVIEW

File: knotx-core/build.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+    inputs.properties(tokens)
+
+    from("src/main/java-templates") {
+      include("*.java")
+      filter<ReplaceTokens>("tokens" to tokens)
+    }
+    into("src/main/generated/io/knotx")
+  }
+
+  getByName<RatTask>("rat") {
```
Comment (pun-ky) [inline knotx-core/build.gradle.kts]: named<RatTask>()

File: settings.gradle.kts [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+rootProject.name = "knotx-root"
+include(":knotx-core")
+include(":documentation")
+
+project(":knotx-core").projectDir = file("$rootDir/knotx-core")
```
Comment (pun-ky) [inline settings.gradle.kts]: Redundant lines, project path is matching directory names

File: knotx-core/build.gradle.kts [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+
+plugins {
+  id("java-library")
+  id("maven-publish")
+  id("org.nosphere.apache.rat") version "0.4.0"
+}
+
+description = "Knot.x Core"
+
+dependencies {
```
Comment (szymon-owczarzak) [inline knotx-core/build.gradle.kts]: You could probably move `deps` into separate file and just apply them here - actual build script would be more readable.

Comment (tomaszmichalak) [reply knotx-core/build.gradle.kts]: fixed.

File: settings.gradle.kts (hunk 2 above)
Comment (tomaszmichalak) [reply settings.gradle.kts]: fixed

**Pull request merged**